docs/types.rst

```diff
@@ -516,7 +516,8 @@ Every complex type, i.e. *arrays* and *structs*, has an additional
 annotation, the "data location", about whether it is stored in memory or in storage. Depending on the
 context, there is always a default, but it can be overridden by appending
 either ``storage`` or ``memory`` to the type. The default for function parameters (including return parameters) is ``memory``, the default for local variables is ``storage`` and the location is forced
-to ``storage`` for state variables (obviously).
+to ``storage`` for state variables (obviously). As of version 0.5.0, data locations for function parameters and function return parameters no longer have a default location, and ``memory`` or ``storage``
```

Comment:
Could you please rephrase this, not talking about defaults at all and then adding a ``.. note:`` that prior to version 0.5.0 the data location could be omitted defaulting on different locations depending on the kind of variable or parameter (something like that) - so it does not even have to list the exact logic.

Reply:
I have changed the wording.